File: csharp-project-euler-100/src/P005.cs


namespace net.talaatharb.projecteuler100
{

    /**
    * @author mharb
    *
    */
    public class P005 : Solution
    {

        /**
         * Problem solution entry point
         * 
         * @param args
         */
        public static void Main(string[] args)
        {
            new P005().run();
        }

        public override long solve()
        {
            problemNumber = 5;
            const int last = 20;
            List<int> sequence = new List<int>();
            int result = 1;
            int number = 2;
            int counter = number;
            while (counter <= last)
            {
                foreach (int i in sequence)
                {
                    if (number % i == 0)
                    {
                        number /= i;
                    }
                }

                result *= number;
                sequence.Add(number);
                counter += 1;
                number = counter;
            }

            return result;
        }
    }
}

File: rust-project-euler-100/src/solutions/p005.rs
use crate::Solvable;

pub struct P005;
impl Solvable for P005 {
    fn problem_name(&self) -> &str { "p005" }
    fn solve(&self) -> i64 {
        let last = 20i64;
        let mut result = 1i64;
        let mut sequence: Vec<i64> = Vec::new();
        let mut number = 2i64;
        let mut counter = number;
        while counter <= last {
            let mut n = number;
            for &j in &sequence {
                if n % j == 0 {
                    n /= j;
                }
            }
            result *= n;
            sequence.push(n);
            counter += 1;
            number = counter;
        }
        result
    }
}
